pkg/eks/auth.go

```diff
@@ -186,13 +187,13 @@ func (c *ClusterProvider) NewClientConfig() (*ClientConfig, error) {
 	return clientConfig, nil
 }
 
-func (c *ClientConfig) WithExecHeptioAuthenticator() *ClientConfig {
+func (c *ClientConfig) WithExecAuthenticator() *ClientConfig {
 	clientConfigCopy := *c
 
 	x := clientConfigCopy.Client.AuthInfos[c.ContextName]
 	x.Exec = &clientcmdapi.ExecConfig{
 		APIVersion: "client.authentication.k8s.io/v1alpha1",
-		Command:    "heptio-authenticator-aws",
+		Command:    utils.DetectAuthenticator(),
```

Comment:
Nice, its great kops has this already.

Reply:
This is in our utils package, which now wraps the single kops utils function we use.